common/libs/transaction.ts

```diff
@@ -71,13 +71,63 @@ export function getTransactionFields(tx: EthTx) {
   };
 }
 
-export async function generateCompleteTransactionFromRawTransaction(
+function getValue(token: Token | null | undefined, tx: ExtendedRawTransaction) {
+  let value;
```

Comment:
Can make value a constant, with its value being the return value of a ternary. Would make it more legible.

Reply:
`const value =  new Big((token ? ERC20.$transfer(tx.data): tx).value)`
Or split it into two lines to avoid the parentheses